lib/presentation/workout_actions/workout_exercise_card.dart

```diff
@@ -0,0 +1,102 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:stronk/domain/constants.dart' as Constants;
+import 'package:stronk/domain/model/workout.dart';
+import 'package:stronk/presentation/workout_actions/workout_action_bloc.dart';
+import 'package:stronk/presentation/workout_actions/workout_action_exerciseSetDisplay.dart';
+
+enum workoutOptions { editWorkoutName, editWorkoutDescription }
+
+class WorkoutExerciseCard extends StatelessWidget {
+  final WorkoutExercise workoutExercise;
+  final String workoutId;
+  final WorkoutActionBloc bloc;
+
+  WorkoutExerciseCard(
+      {@required this.workoutExercise,
+      @required this.workoutId,
+      @required this.bloc});
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+        width: 120,
+        margin: EdgeInsets.symmetric(horizontal: 2),
+        child: InkWell(
+          onTap: () => {
+            Navigator.push(
+                context,
+                MaterialPageRoute(
+                    builder: (context) => WorkoutExercisesSetPage(
+                        workoutId: workoutId,
+                        workoutExerciseId: workoutExercise.id,
+                        exerciseSet: workoutExercise.exerciseSets,
+                        bloc: bloc)))
+          },
+          child: Card(
+            child: Column(
+                mainAxisSize: MainAxisSize.min,
+                mainAxisAlignment: MainAxisAlignment.start,
+                children: <Widget>[
+                  ListTile(
+                    trailing: PopupMenuButton<workoutOptions>(
```

Comment:
Maybe we could move this to a function since it makes the `build` fun harder to read ;P

Reply:
Yeah I was thinking the same thing.. I'll be sure to do it the next commit and I think I'll need to add more tests specifically for workout exercises